CaryTick_MonthlyMet.R

```diff
@@ -0,0 +1,238 @@
+library(rjags)
+
+t <- read.csv("tick_cleaned")
+
+t <- subset(t, Grid == "Green Control")
+
+ymd <- as.data.frame(strsplit(as.character(t$DATE),"-"))
+yr_mon.index <- t(ymd[-3,])
+yr_mon.index <- paste(yr_mon.index[,1], yr_mon.index[, 2], sep = "")
+yr_mon.index <- as.integer(yr_mon.index)
+
+yr_mon.index <- transform(yr_mon.index,id=as.numeric(factor(yr_mon.index)))
+yr_mon.index <- yr_mon.index[,2]
+
+t <- t[,c("n_larvae", "n_nymphs", "n_adults")]
+dat.t <- t(t)
+
+### average monthly
+met <- read.csv("Met_Cary")
+met <- met[, c("DATE","MAX_TEMP","MAX_RH","TOT_PREC")]
+met.y <- as.data.frame(strsplit(as.character(met$DATE),"-"))
+met.y <- t(met.y[-3,])
+met.y <- paste(met.y[,1], met.y[, 2], sep = "")
+met <- cbind(met.y, met[,-1])
+colnames(met) <- c("yr_mon","MAX_TEMP","MAX_RH","TOT_PREC")
+yr_mon <- unique(met.y)
+
+mon.met <- data.frame()
+for(i in 1:length(yr_mon)){
+  xx <- met[met$yr_mon == yr_mon[i], -1]
+  xx <- apply(xx, 2, mean, na.rm = TRUE)
+  mon.met <- rbind(mon.met, xx)
+}
+mon.met <- cbind(yr_mon, mon.met)
+colnames(mon.met) <- colnames(met)
+
+precip <- mon.met[, "TOT_PREC"] ## mean annual precip
+temp <- mon.met[, "MAX_TEMP"] ## mean annual temperature
+rh <- mon.met[, "MAX_RH"] ## mean annual relative humidity
+
+precip <- scale(precip, scale = FALSE)
+temp <- scale(temp, scale = FALSE)
+rh <- scale(rh, scale = FALSE)
+
+precip <- precip[,1]
+temp <- temp[,1]
+rh <- rh[,1]
+
+R <- diag(1, 3, 3)
+
+data <- list(y = dat.t, 
+             R = R,
+             time = ncol(dat.t),
+             yr_mon.index = yr_mon.index,
+             temp = temp,
+             precip = precip,
+             rh = rh,
+             b0 = as.vector(c(0, 0, 0)),   # regression beta means
+             Vb = solve(diag(10000,3)))   # regression beta precisions
+
+noise <- function(dat, sd){
```

Comment:
There's too much in the PR for me to follow everything, but I feel like there is a **LOT** of redundant code in this repo. If you're reusing functions, define them once in their own files. If you're writing the same code again and again (or worse, copy-and-pasting) then you should figure out what parts are repeated and put them in functions. For example, I think a lot of your files contain code that is identical except for small differences in JAGS models -- to me that seems like something better handled by having one bit of code that handles the I/O for a specific class of models and then pass different individual models to that code.